storm-server/src/main/java/org/apache/storm/blobstore/LocalFsBlobStoreFile.java

```diff
@@ -72,6 +74,14 @@ public String getKey() {
         return key;
     }
 
+    @Override
+    public long getVersion() throws IOException {
+        try (FileInputStream fis = new FileInputStream(path)) {
+            byte[] bytes = DigestUtils.sha1(fis);
```

Comment:
Thank you for the suggestion, I've update it to Sha256

Reply:
Do we have a fealing how often `getVersion(...)` is called? Creating a SHA hash is rather expensive compared to the modification date (just hink, if we need to do caching after first call or a like) ?

Reply:
This is being ran by the AsyncLocalizer every interval defined by _supervisor.localizer.update.blob.interval.secs_  but this won't have impact on the worker but on the supervisor. We wouldn't need to cache it but nevertheless we can add cache.

Reply:
Since this is being ran continuously we can opt for a Murmur has that will prioritise fast hashing(suggested by @reiabreu) and this way we would opt for not using caching

Reply:
Hey, after a brief discussion, we've decided to follow with Checksum instead of Murmur since Checksum computation is faster. [Commit](https://github.com/apache/storm/pull/3697/commits/c135adc0f3adcb430fff5e42b7a5a2771f0c0969) with the changes.